src/engine/plugins/match.cpp

```diff
@@ -139,6 +139,15 @@ Status MatchPlugin::HandleRequest(const RoutingAlgorithmsInterface &algorithms,
         return Error("InvalidValue", "Invalid coordinate value.", json_result);
     }
 
+    std::for_each(parameters.radiuses.begin(), parameters.radiuses.end(), [&](auto &radius) {
+        // enforce maximum size of radius in radiuses for performance reasons
+        if (max_radius_map_matching > 0.0 && radius > max_radius_map_matching)
+        {
+            return Error(
+                "TooBig", "Radius search size is too large for map matching.", json_result);
+        }
```

Comment:
This does not compile - you're inside the `for_each`'s lambda and are trying to return an error from the lambda here.

What you can do instead of your `for_each` is check with `any_of` if any radius is invalid and if so return an error.

http://en.cppreference.com/w/cpp/algorithm/all_any_none_of